pkg/controller/datavolume/util.go

```diff
@@ -44,85 +44,105 @@ const (
 )
 
 // renderPvcSpec creates a new PVC Spec based on either the dv.spec.pvc or dv.spec.storage section
-func renderPvcSpec(client client.Client, recorder record.EventRecorder, log logr.Logger, dv *cdiv1.DataVolume) (*v1.PersistentVolumeClaimSpec, error) {
+func renderPvcSpec(client client.Client, recorder record.EventRecorder, log logr.Logger, dv *cdiv1.DataVolume, pvc *v1.PersistentVolumeClaim) (*v1.PersistentVolumeClaimSpec, error) {
 	if dv.Spec.PVC != nil {
 		return dv.Spec.PVC.DeepCopy(), nil
-	}
-
-	if dv.Spec.Storage != nil {
-		return pvcFromStorage(client, recorder, log, dv)
+	} else if dv.Spec.Storage != nil {
+		return pvcFromStorage(client, recorder, log, dv, pvc)
 	}
 
 	return nil, errors.Errorf("datavolume one of {pvc, storage} field is required")
 }
 
-func pvcFromStorage(client client.Client, recorder record.EventRecorder, log logr.Logger, dv *cdiv1.DataVolume) (*v1.PersistentVolumeClaimSpec, error) {
-	storage := dv.Spec.Storage
-	pvcSpec := copyStorageAsPvc(log, storage)
+func pvcFromStorage(client client.Client, recorder record.EventRecorder, log logr.Logger, dv *cdiv1.DataVolume, pvc *v1.PersistentVolumeClaim) (*v1.PersistentVolumeClaimSpec, error) {
+	var pvcSpec *v1.PersistentVolumeClaimSpec
+
+	if pvc == nil {
+		pvcSpec = copyStorageAsPvc(log, dv.Spec.Storage)
+		if err := renderPvcSpecVolumeModeAndAccessModes(client, recorder, log, dv, pvcSpec); err != nil {
+			return nil, err
+		}
+	} else {
+		pvcSpec = pvc.Spec.DeepCopy()
+	}
+
+	if err := renderPvcSpecVolumeSize(client, dv.Spec, pvcSpec); err != nil {
+		return nil, err
+	}
+
+	return pvcSpec, nil
+}
 
+func renderPvcSpecVolumeModeAndAccessModes(client client.Client, recorder record.EventRecorder, log logr.Logger, dv *cdiv1.DataVolume, pvcSpec *v1.PersistentVolumeClaimSpec) error {
```

Comment:
Maybe `renderVolumeAndAccessModes`? I think we could remove the pvcSpec from the name, but `renderPvcSpecVolumeAndAccessModes` would also work

Reply:
renderVolumeAndAccessModes sounds good to me.